modules/elasticsearch/src/main/java/org/testcontainers/elasticsearch/ElasticsearchContainer.java

```diff
@@ -63,4 +77,17 @@ public String getHttpHostAddress() {
     public InetSocketAddress getTcpHost() {
         return new InetSocketAddress(getContainerIpAddress(), getMappedPort(ELASTICSEARCH_DEFAULT_TCP_PORT));
     }
+
+    /**
+     * If your Elasticsearch container has shards, wait until the shards are started
+     *
+     */
+    public ElasticsearchContainer waitingForShards() {
```

Comment:
Why not part of initializing? A vanilla ES has no shards so no added delay. One with preloaded data will want to wait. Maybe turn this behavior if required with flag

Reply:
This is tricky, how do I know the cointainer is not empty? For know, I would say, let the developer decide if he wants to wait for the shards to be up or not.

Reply:
I could explore that the default waiting behaviour checks for the shards, if the container has not preload data, the `/_cat/shards` endpoint should return a not found or not available. Only if it return a valid response I could trigger the waiting for shards code.